### ISSUE TITLE
React Prototype Requirements

### ISSUE THREAD

**Issue by DavidLiedle** [ezsystems/specs#6]
Awaiting discussion with PM/Engineering

**Comment by sunpietro** [comment]
Using view libraries like React.js or Vue.js might be a reasonable idea. They allows to build new views quickly and keep rich UI interactions we have in Studio.
We have to keep in mind that, either React.js or Vue.js will need some sort of state management and routes management. 
In terms of React.js, the state management can be help by Redux.js and routing by `react-route-redux` tool.

**Comment by DavidLiedle** [comment]
Indeed. Given that React is a library, it requires a lot of additional stuff to do all the things. Angular, by contrast, being a framework provides the structure and maturity to expose functionality in a unified "world".

Regardless, creating a prototype repo will give us a playground to answer questions we won't know how to ask until we get our hands dirty getting a better picture of what it can do in our world. For the same reasons given in #7 it should be examined from both directions as well:

1. What would it look like to build Studio in React?
2. What would it look like for React developers to extend Studio, regardless of what we use to build it?
3. How does it compare with the DX (both internal and external) of the other prototypes?
